stream-chat-android-offline/src/main/java/io/getstream/chat/android/offline/plugin/logic/querychannels/internal/QueryChannelsLogic.kt

```diff
@@ -125,11 +125,26 @@ internal class QueryChannelsLogic(
      *
      * @param channel [Channel]
      */
-    internal suspend fun addChannel(channel: Channel) {
+    private suspend fun addChannel(channel: Channel) {
         addChannels(listOf(channel), repos)
         logicRegistry.channel(channel.type, channel.id).updateDataFromChannel(channel)
     }
 
+    /**
+     * Calls watch channel and adds result to the query.
+     *
+     * @param cid cid of the channel.
+     */
+    private suspend fun watchAndAddChannel(cid: String) {
+        val result = client.channel(cid = cid).watch().await()
+
+        logger.logE(result.isSuccess.toString())
+        if (result.isSuccess) {
+            logger.logE(result.data().toString())
+            addChannel(result.data())
+        }
```

Comment:
Hey @adasiewiczr,

Should we log the success result with an error message?

Reply:
good catch!